python/cudf/cudf/_lib/cpp/io/types.pxd

```diff
@@ -72,6 +72,7 @@ cdef extern from "cudf/io/types.hpp" \
         column_in_metadata& set_int96_timestamps(bool req)
         column_in_metadata& set_decimal_precision(uint8_t precision)
         column_in_metadata& child(size_type i)
+        column_in_metadata& set_output_as_binary(bool binary)
```

Comment:
A minor issue to address with all of the methods of this class (out of scope for this PR), none of them are marked as `except +` so users will just see a C++ abort if anything fails. Most of them shouldn't fail since they're just setters of course, but I could see `set_decimal_precision` throwing for an invalid precision or `set_name` having some restriction on the names.

Reply:
Good catch. Most of these can be made `noexcept`, but that's not the case as of now.

Reply:
https://github.com/rapidsai/cudf/issues/11399